src/sinks/http.rs

```diff
@@ -369,6 +395,26 @@ fn validate_headers(
     Ok(headers)
 }
 
+fn validate_payload_wrapper(
+    payload_prefix: &str,
+    payload_suffix: &str,
+    encoder: &Encoder<Framer>,
+) -> crate::Result<(String, String)> {
+    let payload = [payload_prefix, "{}", payload_suffix].join("");
+    match (encoder.serializer(), encoder.framer()) {
+        (
+            Serializer::Json(_),
+            Framer::CharacterDelimited(CharacterDelimitedEncoder { delimiter: b',' }),
+        ) => match serde_json::from_str::<serde_json::Value>(&payload) {
+            Ok(_) => Ok((payload_prefix.to_owned(), payload_suffix.to_owned())),
+            Err(_) => {
+                Err("Payload prefix and suffix wrapper must produce a valid JSON object.".into())
+            }
+        },
+        _ => Ok((payload_prefix.to_owned(), payload_suffix.to_owned())),
+    }
```

Comment:
I believe you can collapse the nested match into one match here:

```suggestion
    match (encoder.serializer(), encoder.framer(), serde_json::from_str::<serde_json::Value>(&payload) ) {
        (
            Serializer::Json(_),
            Framer::CharacterDelimited(CharacterDelimitedEncoder { delimiter: b',' }),
            Err(_)
         ) => Err("Payload prefix and suffix wrapper must produce a valid JSON object.".into()),
        _ => Ok((payload_prefix.to_owned(), payload_suffix.to_owned())),
    }
```

Reply:
ah nice, i thought the nested match looked not so nice, but didn't come up with the idea to collapse it.